### ISSUE TITLE
Have a setting for enabling / disabling automatic exploration of patch in Max / Pd

### ISSUE THREAD

**Issue by jcelerier** [OSSIA/libossia#369]
eg to make stuff faster on loading, it would be nice to have the ability to disable automatic exploration: this way it would take much less time during the setup.

**Comment by jln-** [comment]
FWIW, I'm not sure I get what you mean by automatic exploration. What is that to say ?

**Comment by jcelerier** [comment]
currently objects are created automatically in the tree (eg ossia.parameter, etc... creates object). But when loading with a lot of ossia.parameter it already takes multiple minutes at loading. If instead there was no automatic exploration during the loading of the patch, and the exploration happened only after the patch had loaded, it would be quite faster.

**Comment by jln-** [comment]
Ah ok ok. Thanks. Yes, it makes perfect sense !

**Comment by bltzr** [comment]
yes, that would be the more sensible way to do it... but AFAIK there is no reliable way in Max to know that a patcher has been loaded (my usual hack is to used [loadbang] with a variable number of [deferlow]s and pray that it will work)

**Comment by jcelerier** [comment]
> but AFAIK there is no reliable way in Max to know that a patcher has been loaded

hence why it would have to be manual :p

**Comment by bltzr** [comment]
You mean that the user would have to click a button for that ?

**Comment by jcelerier** [comment]
yes

**Comment by avilleret** [comment]
@jcelerier Can you share such a patch that takes minutes to load ?

**Comment by avilleret** [comment]
At first the registration in Max was delayed by 1 tick to let the patcher load before starting registration.
Hopefully, this reduces loading time.
I never receive benchmark but one patcher from @jcelerier.
So I'm curious how long it takes to load more than 1000 parameters.
I had some loading time issue in Pd 2 weeks ago, but the cause was not very clear.
A patcher that loaded in more than 5 minutes now load in less than 100ms.
One difference between Pd and Max : now in Pd each object registrer its hierarchy, all its parents are stored into a `std::vector<t_canvas*>`.
Thus instead of iterating over all patcher's object to find a friend, one has to iterate over all ossia objects and check its hierarchy.
This could be implemented in Max too, but some benchmark are needed to measure how efficient this could be.
So, again, @jln- @matcham @bltzr  : please make big patchers and share them. 
They don't need to do fancy thnigs, just containing a lots of (ossia and non-ossia) objects is enough.

**Comment by jln-** [comment]
FWIW, parameters registration itself is not that slow, to me. Just made a quick test patch with 1000 params and patch load in a couple of seconds or so.

Of course, that would be different if each param was bound to some processes initializing stuff but still.

<pre><code>
----------begin_max5_patcher----------
12220.3ocwcurjbkcddFdL4UAhVZjCZF4Zs9Osrmvq.GgG6vgiljPTPAIPGM
PKSZE5d23PCJQwpo.a99UeSppqrRr2+46dmGd5UUY8u7S+Ie0u7M+9W91u5E
+2dw+qW7S9I+K+zexO4iWzGtfex2+0+ju5280+9e0u8qe6GuZe0u6ku8se8u
4ke0O6Seu28xe+693k+5u98emu4q+U+wuyae2e329xO9s97k7Me869U+iu50
+l+Oe6K+Uu6S6y0ib+ye7ydwYme3S85Ceb+9K5E+u+9+Qu969cu50+1W9tOt
22+aW3a9t284Kc88W5q90eb28le4+z+04y6yOc0d2e3ad4m1ge0W8G2xey29
x29xW+tu9cu5Mu9Kaj9v+t+0e5O8Ce3m8ElqW+x+uued9yp0ad6ae0W+iqT4
iO9o9uboVeokZATp+7QhqTu6Mu8O769ku429WUrpJ93DUebjRrVU+3i0O7Hw
0pu4ae0qe2OtyplOcHb+ioTOdhRke0W34N+46Xtd7e486l+pxwcOeXNV8Guq
+4AziFshevya9vAr+yN04Gdp3R0+zad0qew4Gycxh39wO8eRrNeww5727cxd
hQ5GQo9ke26d2ad8eUIYx5iOmwGOdsiO+w+VeXm8OXQ9ke8q+M+mVkevoh6z
mu6+2qd+80d73uxZc9340mOddcF+Xt214Ix0i+x45m8u84uj688COk+Hx2Oz
qd5iudfe9270e66eUTu6ke6K9c+g+Oe7K94eHpu3W7gaBu3e329lu9cu3W7s
ue1e4Kd7yeeu+4u3W7qe4+vW+c+1289uds+i2w429pW+xe0a9tW+wsc7T27d
xqwWzAs3SmLE6OdmqU7w3T+s+Xj2e72q+uvHwcF9W+MeyKe8u9u5CE+07Ho6
O8vV6l50qz+M+HoO0HQ95Ud4Gi5+gy9+Q8R8VqOcveC0NfWp2SLRbs6W8le2
u782I9e6QJdwq9vYme6q9M+lW9su8EqeLQbcqO8o7GyCEu+yiX9CGwe1WdHe
pwB7wde6u5ae027tW7oN+9882+e89X99qwK96Wu3W7GK127l29pOLiuX+3E+
8+ntOdt+zK6E60VF+Med5SMRf9s+wW81Olieb2uNyO9pVpeTOj3SbF4F4Lxe
3o5se8+7m2f+c+O9vl7O41+K9p+6e3C+c+O+vGe+i78oKA++wBO4KjPySc8o
my8iOi64FPGjVOV8eaGm9gFqmgPuUF5u+1EYoGhR+Dy0yPpOROmlO0WjSpsj
5PYpC7TuePj5vRpSkoN4S8hH0okTWJScwm5MQpKKotUl5lO0GhT2VR8nL0Ce
pChTOVR8UYpu7oNIR80RpWOT150iu+Y7evE6hH1O0f8bTaodw0hu1HhwmZvd
NpsTz3ZyWaD03SMXOG0Vpabc3qMBb7oFrmiZKkNtB7ZePriO0f8bTao5wUxW
aD93SMXOG0VJfbU70FQP9TC1yQskZHWMesQPjO0f8bTaoLx0vWaDG4SMXOG0
Vpjbc4qMBk7oFrmi0n4g1Eogu1Eypz3o1RsjadK4AwRt8XI2ZW.RdK4gYEH8
XI2RsjadK4AwRt8XI2RsjadKYfXI2drjaoVxMukLPrjaOVxsTK4l2RFHVxsG
K4VpkbyaICDK41ikbK0Rt4sjAhkb6wRtkZI27Vx.wRt8XIOOz9SgFesKleLz
7Tas+brxaICleRV8XIORsjGdKYfXIOdrjGs+7rxaIClefV8XIORsjGdKYhXI
OdrjGoVxCukLQrjGOVxiTK4g2RlHVxiGK4Qpk7vaISDK4wik7H0Rd3sjIhk7
3wRdjZIO7VxDwRd7XIiGZ+0rgu1EyumMdpsTKYvaISDKY3wRFRsjAukLQrjg
GKYH0RF7VxDwRFdrjg1e8H4sjEyuejdrjgTKYvaIKDKY3wRFRsjAukrPrjgG
KYH0RF7VxBwRFdrjgTKYvaIKDKY3wRFRsjAukrPrjgGKY9P66i.70tXdiDvS
skZISdKYgXISOVxTpkL4sjEhkL8XISoVxj2RVHVxzikLkZISdKYiXISOVxT6
61N7Vxl4saGOVxTpkL4sjMhkL8XISoVxj2R1HVxzikLkZISdKYiXISOVxTpk
L4sjMhkL8XIqGZeiRiu1Ey6TZdpsTKYwaIaDKY4wRVRsjEukrQrjkGKYI0RV
7VxFwRVdrjkTKYwaIGDKY4wRVRsjEukbPrjkGKYo8MuUdK4v7t2pGKYI0RV7
VxAwRVdrjkTKYwaIGDKY4wRVRsjEukbPrjkGKY+P66Dz70tXdqf1SskZIadK
4fXIaOVxVpkr4sjChkr8XIaoVxl2RNHVx1ikrkZIadK4EwR1drjsTKYyaIuH
Vx1ikrkZIadK4EwR1drjs1+VfvaIuL+w.wikrkZIadK4EwR1drjsTKYyaIuH
Vx1ikbdn8O0M70tX9acimZK0RN7VxKhkb7XIGoVxg2RdQrjiGK4H0RN7VxKh
kb7XIGoVxA2RtdfXIGOVxQpkbR9ZiXIGOVxQpkbJ9ZiXIGOVxQpkbZ9ZiXIG
OVxQ6eZIG9Zy72VROVxQpkbt70FwRNdrj2GZ+a4IesKl+Xd5o1Rsj2EesQrj
WOVxqTK4cyWaDK40ik7J0RdO70FwRd8XIuRsjWdK4BwRd8XIuRsjWdK4BwRd
8XIuRsjWdK4BwRd8XIuRsjWdK4BwRd8XIuRsjWdK4BwRd8XIuRsjWdK4BwRd
8XIWOjhIe+l+6uYMb8FQS9jS1yRvWZC9hO3MSvWlB9Vav27AeXB91TvOZC9g
O3WlfeLE7PavC7fuevD7vTvSsAO4C9hI3oofWZCdwG7MSvKSAu0F7lO3Glf2
lB9nM3CevClfOlB9Uavu7AOYB90SvWZklqO64JtfyHMepI6YI3ZklqEevYjl
O0j8rDbsRy0lO3LRymZxdVBtVo45vGbFo4SMYOKAWqzbE3A+vHMepI6YI3Zk
lqjO3LRymZxdVBtVo4p3CNiz7olrmkfqUZtZ9fyHMepI6YI3ZklqgO3LRymZ
xdVBtVo45xGbFo4SMYOGAeqUZt4WSyCizbaZMM2Zkla90z7vHM2lVSysVo4l
eMMOLRyso0zbqUZt4WSyCizbaZMM2Zkla90zLXjlaSqo4VqzbyullAizbaZM
M2Zkla90zLXjlaSqo4VqzbyullAizbaZMM2Zkla90zLXjlaSqo4VqzbyullA
izbaZMMOZklGdoYvHMOljlGsRyCuzLXjlGSRyiVo4gWZFLRyiIo4Qqz7vKMC
Fo4wjz7nUZd3klIiz7XRZdzJMO7RyjQZdLIMOZklGdoYxHMOljlGsRyCuzLY
jlGSRyiVo4gWZlLRyiIo4Qqz7vKMSFo4wjzLzJMCdoYxHMCSRyPqzL3klIiz
LLIMCsRyfWZlLRyvjzLzJMCdoYxHMCSRyPqzL3klEizLLIMCsRyfWZVLRyvj
zLzJMCdoYwHMCSRyPqzL3klEizLLIMCsRyfWZVLRyvjzLzJMCdoYwHMCSRyT
qzL4klEizLMIMSsRyjWZVLRyzjzL0JMSdoYwHMSSRyTqzL4klEizLMIMSsRy
jWZ1LRyzjzL0JMSdoYyHMSSRyTqzL4klMizLMIMSsRyjWZ1LRyzjzL0JMSdo
YyHMSSRyTqzL4klMizLMIMKsRyhWZ1LRyxjzrzJMKdoYyHMKSRyRqzr3klMi
zrLIMKsRyhWZ1LRyxjzrzJMKdo4vHMKSRyRqzr3klCizrLIMKsRyhWZNLRyx
jzrzJMKdo4vHMKSRyRqzr3klCizrLIMKsRyhWZNLRyxjzr0JMado4vHMaSRy
Vqzr4klCizrMIMasRylWZNLRy1jzr0JMado4vHMaSRyVqzr4klWFoYaRZ1Zk
lMuz7xHMaSRyVqzr4klWFoYaRZ1ZklMuz7xHMaSRyVqzr4klWFoYaRZ1ZklM
uz7xHMaSRyQqzb3klWFo4XRZNZklCuz7xHMGSRyQqzb3klWFo4XRZNZklCuz
7xHMGSRyQqzbvkl6GLRywjzbzJMmjO3LRywjzbzJMmhO3LRywjzbzJMmlO3L
RywjzbzJMmgO3LRywjzbzJMmKevYjliIo4Uqz79fO3LRyqIo4Uqz7t3CNiz7
ZRZd0JMua9fyHMuljlWsRy6gO3LRyqIo4Uqz7xKMWLRyqIo4Uqz7xKMWLRyq
Io4Uqz7xKMWLRyqIo4Uqz7xKMWLRyqIo4Uqz7xKMWLRyqIo4Uqz7xKMWLRyq
Go49gTo46277AGQZ9jS1yRvWZCNuzb0LAeYJ3asAmWZtFlfuME7i1fyKMWWl
feLE7Pav4kl6GLAOLE7Tav4kl6ESvSSAuzFbdo4dyD7xTvasAmWZtOLAuME7
Qav4kl6fI3iofe0Fbdo4NYBtIo4RqzbwKM2LRykIo4Rqzb8YOWwEbFo4SMYO
KAWqzbwKM2LRykIo4RqzbwKM2LRykIo4RqzbwKMOLRykIo4RqzbwKMOLRykI
o4RqzbwKMOLRykIo4RqzbwKMOLRykIo4RqzbwKMOLRykIo4RqzbwKMOLRykI
o4VqzbyKMOLRysIo4VqzbyullGFo41zZZt0JM27RyCizbaRZt0JM27RyCizb
aRZt0JM27RyfQZtMIM2ZkladoYvHM2ljlasRyMuzLXjlaSRysVo4lWZFLRys
Io4VqzbyKMCFo41jzbqUZt4klAizbaRZdzJMO7RyfQZdLIMOZklGdoYvHMOl
jlGsRyCuzLXjlGSRyiVo4gWZFLRyiIo4Qqz7vKMSFo4wjz7nUZd3klIiz7XR
ZdzJMO7RyjQZdLIMOZklGdoYxHMOljlGsRyCuzLYjlGSRyiVo4gWZlLRyiIo
YnUZF7RyjQZFljlgVoYvKMSFoYXRZFZklAuzLYjlgIoYnUZF7RyjQZFljlgV
oYvKMKFoYXRZFZklAuzrXjlgIoYnUZF7RyhQZFljlgVoYvKMKFoYXRZFZklA
uzrXjlgIoYnUZF7RyhQZFljloVoYxKMKFoYZRZlZklIuzrXjloIoYpUZl7Ry
hQZlljloVoYxKMKFoYZRZlZklIuzrYjloIoYpUZl7RylQZlljloVoYxKMaFo
YZRZlZklIuzrYjloIoYpUZl7RylQZlljloVoYxKMaFoYZRZVZklEuzrYjlkI
oYoUZV7RylQZVljlkVoYwKMaFoYYRZVZklEuzrYjlkIoYoUZV7RygQZVljlk
VoYwKMGFoYYRZVZklEuzbXjlkIoYoUZV7RygQZVljlkVoYwKMGFoYYRZVZkl
EuzbXjlkIoYqUZ17RygQZ1ljlsVoYyKMGFoYaRZ1ZklMuzbXjlsIoYqUZ17R
ygQZ1ljlsVoYyKMuLRy1jzr0JMado4kQZ1ljlsVoYyKMuLRy1jzr0JMado4k
QZ1ljlsVoYyKMuLRy1jzr0JMado4kQZ1ljliVo4vKMuLRywjzbzJMGdo4kQZ
NljliVo4vKMuLRywjzbzJMGdo4kQZNljliVo4fKMOOXjliIo4nUZNIevYjli
Io4nUZNEevYjliIo4nUZNMevYjliIo4nUZNCevYjliIo4nUZNW9fyHMGSRyq
Vo48AevYjlWSRyqVo4cwGbFo40jz7pUZd27AmQZdMIMuZkl2CevYjlWSRyqV
o4kWZtXjlWSRyqVo4kWZtXjlWSRyqVo4kWZtXjlWSRyqVo4kWZtXjlWSRyqV
o4kWZtXjlWSRyqVo4kWZtXjlWORyyCoRy2u44CNhz7ImrmkfuzFbdo4pYB9x
Tv2ZCNuzbMLAeaJ3GsAmWZttLA+XJ3g1fyKM2OXBdXJ3o1fyKM2KlfmlBdoM
37Ry8lI3kof2ZCNuzbeXBdaJ3i1fyKM2ASvGSA+pM37RycxDbSRykVo4hWZt
YjlKSRykVo45ydthK3LRymZxdVBtVo4hWZtYjlKSRykVo4hWZtYjlKSRykVo
4hWZdXjlKSRykVo4hWZdXjlKSRykVo4hWZdXjlKSRykVo4hWZdXjlKSRykVo
4hWZdXjlKSRykVo4hWZdXjlKSRysVo4lWZdXjlaSRysVo4leMMOLRyso0zbq
UZt4klGFo41jzbqUZt4klGFo41jzbqUZt4klAizbaRZt0JM27RyfQZtMIM2Z
kladoYvHM2ljlasRyMuzLXjlaSRysVo4lWZFLRysIo4VqzbyKMCFo41jz7nU
Zd3klAiz7XRZdzJMO7RyfQZdLIMOZklGdoYvHMOljlGsRyCuzLXjlGSRyiVo
4gWZlLRyiIo4Qqz7vKMSFo4wjz7nUZd3klIiz7XRZdzJMO7RyjQZdLIMOZkl
GdoYxHMOljlGsRyCuzLYjlGSRyPqzL3klIizLLIMCsRyfWZlLRyvjzLzJMCd
oYxHMCSRyPqzL3klIizLLIMCsRyfWZVLRyvjzLzJMCdoYwHMCSRyPqzL3klE
izLLIMCsRyfWZVLRyvjzLzJMCdoYwHMCSRyPqzL3klEizLLIMSsRyjWZVLRy
zjzL0JMSdoYwHMSSRyTqzL4klEizLMIMSsRyjWZVLRyzjzL0JMSdoYyHMSSR
yTqzL4klMizLMIMSsRyjWZ1LRyzjzL0JMSdoYyHMSSRyTqzL4klMizLMIMSs
RyjWZ1LRyzjzrzJMKdoYyHMKSRyRqzr3klMizrLIMKsRyhWZ1LRyxjzrzJMK
doYyHMKSRyRqzr3klCizrLIMKsRyhWZNLRyxjzrzJMKdo4vHMKSRyRqzr3kl
CizrLIMKsRyhWZNLRyxjzrzJMKdo4vHMKSRyVqzr4klCizrMIMasRylWZNLR
y1jzr0JMado4vHMaSRyVqzr4klCizrMIMasRylWZdYjlsIoYqUZ17RyKizrM
IMasRylWZdYjlsIoYqUZ17RyKizrMIMasRylWZdYjlsIoYqUZ17RyKizrMIM
GsRygWZdYjliIo4nUZN7RyKizbLIMGsRygWZdYjliIo4nUZN7RyKizbLIMGs
RyAWZFOXjliIo4nUZNIevYjliIo4nUZNEevYjliIo4nUZNMevYjliIo4nUZN
CevYjliIo4nUZNW9fyHMGSRyqVo48AevYjlWSRyqVo4cwGbFo40jz7pUZd27
AmQZdMIMuZkl2CevYjlWSRyqVo4kWZtXjlWSRyqVo4kWZtXjlWSRyqVo4kWZ
tXjlWSRyqVo4kWZtXjlWSRyqVo4kWZtXjlWSRyqVo4kWZtXjlWORy3gTo462
77AGQZ9jS1yRvWZCNuzb0LAeYJ3asAmWZtFlfuME7i1fyKMWWlfeLE7Pav4k
l6GLAOLE7Tav4kl6ESvSSAuzFbdo4dyD7xTvasAmWZtOLAuME7Qav4kl6fI3
iofe0Fbdo4NYBtIo4RqzbwKM2LRykIo4Rqzb8YOWwEbFo4SMYOKAWqzbwKM2
LRykIo4RqzbwKM2LRykIo4RqzbwKMOLRykIo4RqzbwKMOLRykIo4RqzbwKMO
LRykIo4RqzbwKMOLRykIo4RqzbwKMOLRykIo4RqzbwKMOLRykIo4VqzbyKMO
LRysIo4VqzbyullGFo41zZZt0JM27RyCizbaRZt0JM27RyCizbaRZt0JM27R
yfQZtMIM2ZkladoYvHM2ljlasRyMuzLXjlaSRysVo4lWZFLRysIo4VqzbyKM
CFo41jzbqUZt4klAizbaRZdzJMO7RyfQZdLIMOZklGdoYvHMOljlGsRyCuzL
XjlGSRyiVo4gWZFLRyiIo4Qqz7vKMSFo4wjz7nUZd3klIiz7XRZdzJMO7Ryj
QZdLIMOZklGdoYxHMOljlGsRyCuzLYjlGSRyiVo4gWZlLRyiIoYnUZF7RyjQ
ZFljlgVoYvKMSFoYXRZFZklAuzLYjlgIoYnUZF7RyjQZFljlgVoYvKMKFoYX
RZFZklAuzrXjlgIoYnUZF7RyhQZFljlgVoYvKMKFoYXRZFZklAuzrXjlgIoY
nUZF7RyhQZFljloVoYxKMKFoYZRZlZklIuzrXjloIoYpUZl7RyhQZlljloVo
YxKMKFoYZRZlZklIuzrYjloIoYpUZl7RylQZlljloVoYxKMaFoYZRZlZklIu
zrYjloIoYpUZl7RylQZlljloVoYxKMaFoYZRZVZklEuzrYjlkIoYoUZV7Ryl
QZVljlkVoYwKMaFoYYRZVZklEuzrYjlkIoYoUZV7RygQZVljlkVoYwKMGFoY
YRZVZklEuzbXjlkIoYoUZV7RygQZVljlkVoYwKMGFoYYRZVZklEuzbXjlkIo
YqUZ17RygQZ1ljlsVoYyKMGFoYaRZ1ZklMuzbXjlsIoYqUZ17RygQZ1ljlsV
oYyKMuLRy1jzr0JMado4kQZ1ljlsVoYyKMuLRy1jzr0JMado4kQZ1ljlsVoY
yKMuLRy1jzr0JMado4kQZ1ljliVo4vKMuLRywjzbzJMGdo4kQZNljliVo4vK
MuLRywjzbzJMGdo4kQZNljliVo4fKMyGLRywjzbzJMmjO3LRywjzbzJMmhO3
LRywjzbzJMmlO3LRywjzbzJMmgO3LRywjzbzJMmKevYjliIo4Uqz79fO3LRy
qIo4Uqz7t3CNiz7ZRZd0JMua9fyHMuljlWsRy6gO3LRyqIo4Uqz7xKMWLRyq
Io4Uqz7xKMWLRyqIo4Uqz7xKMWLRyqIo4Uqz7xKMWLRyqIo4Uqz7xKMWLRyq
Io4Uqz7xKMWLRyqGoY9Ppz78ad9fiHMexI6YI3KsAmWZtZlfuLE7s1fyKMWC
Sv2lB9Qav4klqKSvOlBdnM37Ry8ClfGlBdpM37Ry8hI3oofWZCNuzbuYBdYJ
3s1fyKM2Glf2lB9nM37RycvD7wTvuZCNuzbmLA2jzboUZt3klaFo4xjzboUZ
t9rmq3BNiz7olrmkfqUZt3klaFo4xjzboUZt3klaFo4xjzboUZt3klGFo4xj
zboUZt3klGFo4xjzboUZt3klGFo4xjzboUZt3klGFo4xjzboUZt3klGFo4xj
zboUZt3klGFo4xjzbqUZt4klGFo41jzbqUZt4WSyCizbaZMM2Zklado4gQZt
MIM2Zklado4gQZtMIM2ZkladoYvHM2ljlasRyMuzLXjlaSRysVo4lWZFLRys
Io4VqzbyKMCFo41jzbqUZt4klAizbaRZt0JM27RyfQZtMIMOZklGdoYvHMOl
jlGsRyCuzLXjlGSRyiVo4gWZFLRyiIo4Qqz7vKMCFo4wjz7nUZd3klIiz7XR
ZdzJMO7RyjQZdLIMOZklGdoYxHMOljlGsRyCuzLYjlGSRyiVo4gWZlLRyiIo
4Qqz7vKMSFo4wjzLzJMCdoYxHMCSRyPqzL3klIizLLIMCsRyfWZlLRyvjzLz
JMCdoYxHMCSRyPqzL3klEizLLIMCsRyfWZVLRyvjzLzJMCdoYwHMCSRyPqzL
3klEizLLIMCsRyfWZVLRyvjzLzJMCdoYwHMCSRyTqzL4klEizLMIMSsRyjWZ
VLRyzjzL0JMSdoYwHMSSRyTqzL4klEizLMIMSsRyjWZ1LRyzjzL0JMSdoYyH
MSSRyTqzL4klMizLMIMSsRyjWZ1LRyzjzL0JMSdoYyHMSSRyTqzL4klMizLM
IMKsRyhWZ1LRyxjzrzJMKdoYyHMKSRyRqzr3klMizrLIMKsRyhWZ1LRyxjzr
zJMKdo4vHMKSRyRqzr3klCizrLIMKsRyhWZNLRyxjzrzJMKdo4vHMKSRyRqz
r3klCizrLIMKsRyhWZNLRyxjzr0JMado4vHMaSRyVqzr4klCizrMIMasRylW
ZNLRy1jzr0JMado4vHMaSRyVqzr4klWFoYaRZ1ZklMuz7xHMaSRyVqzr4klW
FoYaRZ1ZklMuz7xHMaSRyVqzr4klWFoYaRZ1ZklMuz7xHMaSRyQqzb3klWFo
4XRZNZklCuz7xHMGSRyQqzb3klWFo4XRZNZklCuz7xHMGSRyQqzbvkl0CFo4
XRZNZklSxGbFo4XRZNZklSwGbFo4XRZNZklSyGbFo4XRZNZklyvGbFo4XRZN
ZklykO3LRywjz7pUZdevGbFo40jz7pUZdW7AmQZdMIMuZkl2MevYjlWSRyqV
o48vGbFo40jz7pUZd4klKFo40jz7pUZd4klKFo40jz7pUZd4klKFo40jz7pU
Zd4klKFo40jz7pUZd4klKFo40jz7pUZd4klKFo40izrdHUZ99MOevQjlO4j8
rD7k1fyKMWMSvWlB9Vav4klqgI3aSA+nM37Ry0kI3GSAOzFbdo49ASvCSAO0
Fbdo4dwD7zTvKsAmWZt2LAuLE7Vav4kl6CSvaSAezFbdo4NXB9XJ3WsAmWZt
SlfaRZtzJMW7RyMizbYRZtzJMWe1yUbAmQZ9TS1yRv0JMW7RyMizbYRZtzJM
W7RyMizbYRZtzJMW7RyCizbYRZtzJMW7RyCizbYRZtzJMW7RyCizbYRZtzJM
W7RyCizbYRZtzJMW7RyCizbYRZtzJMW7RyCizbYRZt0JM27RyCizbaRZt0JM
27qo4gQZtMsllasRyMuz7vHM2ljlasRyMuz7vHM2ljlasRyMuzLXjlaSRysV
o4lWZFLRysIo4VqzbyKMCFo41jzbqUZt4klAizbaRZt0JM27RyfQZtMIM2Zk
ladoYvHM2ljlGsRyCuzLXjlGSRyiVo4gWZFLRyiIo4Qqz7vKMCFo4wjz7nUZ
d3klAiz7XRZdzJMO7RyjQZdLIMOZklGdoYxHMOljlGsRyCuzLYjlGSRyiVo4
gWZlLRyiIo4Qqz7vKMSFo4wjz7nUZd3klIiz7XRZFZklAuzLYjlgIoYnUZF7
RyjQZFljlgVoYvKMSFoYXRZFZklAuzLYjlgIoYnUZF7RyhQZFljlgVoYvKMK
FoYXRZFZklAuzrXjlgIoYnUZF7RyhQZFljlgVoYvKMKFoYXRZFZklAuzrXjl
gIoYpUZl7RyhQZlljloVoYxKMKFoYZRZlZklIuzrXjloIoYpUZl7RyhQZllj
loVoYxKMaFoYZRZlZklIuzrYjloIoYpUZl7RylQZlljloVoYxKMaFoYZRZlZ
klIuzrYjloIoYpUZl7RylQZlljlkVoYwKMaFoYYRZVZklEuzrYjlkIoYoUZV
7RylQZVljlkVoYwKMaFoYYRZVZklEuzbXjlkIoYoUZV7RygQZVljlkVoYwKM
GFoYYRZVZklEuzbXjlkIoYoUZV7RygQZVljlkVoYwKMGFoYYRZ1ZklMuzbXj
lsIoYqUZ17RygQZ1ljlsVoYyKMGFoYaRZ1ZklMuzbXjlsIoYqUZ17RyKizrM
IMasRylWZdYjlsIoYqUZ17RyKizrMIMasRylWZdYjlsIoYqUZ17RyKizrMIM
asRylWZdYjlsIo4nUZN7RyKizbLIMGsRygWZdYjliIo4nUZN7RyKizbLIMGs
RygWZdYjliIo4nUZN3Ry9AizbLIMGsRyI4CNizbLIMGsRyo3CNizbLIMGsRy
o4CNizbLIMGsRyY3CNizbLIMGsRy4xGbFo4XRZd0JMuO3CNiz7ZRZd0JMuK9
fyHMuljlWsRy6lO3LRyqIo4Uqz7d3CNiz7ZRZd0JMu7RyEiz7ZRZd0JMu7Ry
Eiz7ZRZd0JMu7RyEiz7ZRZd0JMu7RyEiz7ZRZd0JMu7RyEiz7ZRZd0JMu7Ry
Eiz75QZ1OjJMe+lmO3HRymbxdVB9Rav4klqlI3KSAeqM37Ry0vD7sofezFbd
o45xD7iofGZCNuzb+fI3gofmZCNuzbuXBdZJ3k1fyKM2alfWlBdqM37Ry8gI
3sofOZCNuzbGLAeLE7q1fyKM2ISvMIMWZklKdo4lQZtLIMWZklqO64JtfyHM
epI6YI3ZklKdo4lQZtLIMWZklKdo4lQZtLIMWZklKdo4gQZtLIMWZklKdo4g
QZtLIMWZklKdo4gQZtLIMWZklKdo4gQZtLIMWZklKdo4gQZtLIMWZklKdo4g
QZtLIM2Zklado4gQZtMIM2Zkla90z7vHM2lVSysVo4lWZdXjlaSRysVo4lWZ
dXjlaSRysVo4lWZFLRysIo4VqzbyKMCFo41jzbqUZt4klAizbaRZt0JM27Ry
fQZtMIM2ZkladoYvHM2ljlasRyMuzLXjlaSRyiVo4gWZFLRyiIo4Qqz7vKMC
Fo4wjz7nUZd3klAiz7XRZdzJMO7RyfQZdLIMOZklGdoYxHMOljlGsRyCuzLY
jlGSRyiVo4gWZlLRyiIo4Qqz7vKMSFo4wjz7nUZd3klIiz7XRZdzJMO7RyjQ
ZdLIMCsRyfWZlLRyvjzLzJMCdoYxHMCSRyPqzL3klIizLLIMCsRyfWZlLRyv
jzLzJMCdoYwHMCSRyPqzL3klEizLLIMCsRyfWZVLRyvjzLzJMCdoYwHMCSRy
PqzL3klEizLLIMCsRyfWZVLRyvjzL0JMSdoYwHMSSRyTqzL4klEizLMIMSsR
yjWZVLRyzjzL0JMSdoYwHMSSRyTqzL4klMizLMIMSsRyjWZ1LRyzjzL0JMSd
oYyHMSSRyTqzL4klMizLMIMSsRyjWZ1LRyzjzL0JMSdoYyHMSSRyRqzr3klM
izrLIMKsRyhWZ1LRyxjzrzJMKdoYyHMKSRyRqzr3klMizrLIMKsRyhWZNLRy
xjzrzJMKdo4vHMKSRyRqzr3klCizrLIMKsRyhWZNLRyxjzrzJMKdo4vHMKSR
yRqzr3klCizrLIMasRylWZNLRy1jzr0JMado4vHMaSRyVqzr4klCizrMIMas
RylWZNLRy1jzr0JMado4kQZ1ljlsVoYyKMuLRy1jzr0JMado4kQZ1ljlsVoY
yKMuLRy1jzr0JMado4kQZ1ljlsVoYyKMuLRy1jzbzJMGdo4kQZNljliVo4vK
MuLRywjzbzJMGdo4kQZNljliVo4vKMuLRywjzbzJMGbo46erWlfaRZNZklSx
GbFo4XRZNZklSwGbFo4XRZNZklSyGbFo4XRZNZklyvGbFo4XRZNZklykO3LR
ywjz7pUZdevGbFo40jz7pUZdW7AmQZdMIMuZkl2MevYjlWSRyqVo48vGbFo4
0jz7pUZd4klKFo40jz7pUZd4klKFo40jz7pUZd4klKFo40jz7pUZd4klKFo4
0jz7pUZd4klKFo40jz7pUZd4klKFo40izbdHUZ99MOevQjlO4j8rD7k1fyKM
WMSvWlB9Vav4klqgI3aSA+nM37Ry0kI3GSAOzFbdo49ASvCSAO0Fbdo4dwD7
zTvKsAmWZt2LAuLE7Vav4kl6CSvaSAezFbdo4NXB9XJ3WsAmWZtSlfaRZtzJ
MW7RyMizbYRZtzJMWe1yUbAmQZ9TS1yRv0JMW7RyMizbYRZtzJMW7RyMizbY
RZtzJMW7RyCizbYRZtzJMW7RyCizbYRZtzJMW7RyCizbYRZtzJMW7RyCizbY
RZtzJMW7RyCizbYRZtzJMW7RyCizbYRZt0JM27RyCizbaRZt0JM27qo4gQZt
MsllasRyMuz7vHM2ljlasRyMuz7vHM2ljlasRyMuzLXjlaSRysVo4lWZFLRy
sIo4VqzbyKMCFo41jzbqUZt4klAizbaRZt0JM27RyfQZtMIM2ZkladoYvHM2
ljlGsRyCuzLXjlGSRyiVo4gWZFLRyiIo4Qqz7vKMCFo4wjz7nUZd3klAiz7X
RZdzJMO7RyjQZdLIMOZklGdoYxHMOljlGsRyCuzLYjlGSRyiVo4gWZlLRyiI
o4Qqz7vKMSFo4wjz7nUZd3klIiz7XRZFZklAuzLYjlgIoYnUZF7RyjQZFljl
gVoYvKMSFoYXRZFZklAuzLYjlgIoYnUZF7RyhQZFljlgVoYvKMKFoYXRZFZk
lAuzrXjlgIoYnUZF7RyhQZFljlgVoYvKMKFoYXRZFZklAuzrXjlgIoYpUZl7
RyhQZlljloVoYxKMKFoYZRZlZklIuzrXjloIoYpUZl7RyhQZlljloVoYxKMa
FoYZRZlZklIuzrYjloIoYpUZl7RylQZlljloVoYxKMaFoYZRZlZklIuzrYjl
oIoYpUZl7RylQZlljlkVoYwKMaFoYYRZVZklEuzrYjlkIoYoUZV7RylQZVlj
lkVoYwKMaFoYYRZVZklEuzbXjlkIoYoUZV7RygQZVljlkVoYwKMGFoYYRZVZ
klEuzbXjlkIoYoUZV7RygQZVljlkVoYwKMGFoYYRZ1ZklMuzbXjlsIoYqUZ1
7RygQZ1ljlsVoYyKMGFoYaRZ1ZklMuzbXjlsIoYqUZ17RyKizrMIMasRylWZ
dYjlsIoYqUZ17RyKizrMIMasRylWZdYjlsIoYqUZ17RyKizrMIMasRylWZdY
jlsIo4nUZN7RyKizbLIMGsRygWZdYjliIo4nUZN7RyKizbLIMGsRygWZdYjl
iIo4nUZN3Ry6CFo4XRZNZklSxGbFo4XRZNZklSwGbFo4XRZNZklSyGbFo4XR
ZNZklyvGbFo4XRZNZklykO3LRywjz7pUZdevGbFo40jz7pUZdW7AmQZdMIMu
Zkl2MevYjlWSRyqVo48vGbFo40jz7pUZd4klKFo40jz7pUZd4klKFo40jz7p
UZd4klKFo40jz7pUZd4klKFo40jz7pUZd4klKFo40jz7pUZd4klKFo40iz78
dKkA+8ad9fiHMexI6YI3KsAmWZtZlfuLE7s1fyKMWCSv2lB9Qav4klqKSvOl
BdnM37Ry8ClfGlBdpM37Ry8hI3oofWZCNuzbuYBdYJ3s1fyKM2Glf2lB9nM3
7RycvD7wTvuZCNuzbmLA2jzboUZt3klaFo4xjzboUZt9rmq3BNiz7olrmkfq
UZt3klaFo4xjzboUZt3klaFo4xjzboUZt3klGFo4xjzboUZt3klGFo4xjzbo
UZt3klGFo4xjzboUZt3klGFo4xjzboUZt3klGFo4xjzboUZt3klGFo4xjzbq
UZt4klGFo41jzbqUZt4WSyCizbaZMM2Zklado4gQZtMIM2Zklado4gQZtMIM
2ZkladoYvHM2ljlasRyMuzLXjlaSRysVo4lWZFLRysIo4VqzbyKMCFo41jzb
qUZt4klAizbaRZt0JM27RyfQZtMIMOZklGdoYvHMOljlGsRyCuzLXjlGSRyi
Vo4gWZFLRyiIo4Qqz7vKMCFo4wjz7nUZd3klIiz7XRZdzJMO7RyjQZdLIMOZ
klGdoYxHMOljlGsRyCuzLYjlGSRyiVo4gWZlLRyiIo4Qqz7vKMSFo4wjzLzJ
MCdoYxHMCSRyPqzL3klIizLLIMCsRyfWZlLRyvjzLzJMCdoYxHMCSRyPqzL3
klEizLLIMCsRyfWZVLRyvjzLzJMCdoYwHMCSRyPqzL3klEizLLIMCsRyfWZV
LRyvjzLzJMCdoYwHMCSRyTqzL4klEizLMIMSsRyjWZVLRyzjzL0JMSdoYwHM
SSRyTqzL4klEizLMIMSsRyjWZ1LRyzjzL0JMSdoYyHMSSRyTqzL4klMizLMI
MSsRyjWZ1LRyzjzL0JMSdoYyHMSSRyTqzL4klMizLMIMKsRyhWZ1LRyxjzrz
JMKdoYyHMKSRyRqzr3klMizrLIMKsRyhWZ1LRyxjzrzJMKdo4vHMKSRyRqzr
3klCizrLIMKsRyhWZNLRyxjzrzJMKdo4vHMKSRyRqzr3klCizrLIMKsRyhWZ
NLRyxjzr0JMado4vHMaSRyVqzr4klCizrMIMasRylWZNLRy1jzr0JMado4vH
MaSRyVqzr4klWFoYaRZ1ZklMuz7xHMaSRyVqzr4klWFoYaRZ1ZklMuz7xHMa
SRyVqzr4klWFoYaRZ1ZklMuz7xHMaSRyQqzb3klWFo4XRZNZklCuz7xHMGSR
yQqzb3klWFo4XRZNZklCuz7xHMGSRyQqzbnkl6GOXjliIo4nUZNIevYjliIo
4nUZNEevYjliIo4nUZNMevYjliIo4nUZNCevYjliIo4nUZNW9fyHMGSRyqVo
48AevYjlWSRyqVo4cwGbFo40jz7pUZd27AmQZdMIMuZkl2CevYjlWSRyqVo4
kWZtXjlWSRyqVo4kWZtXjlWSRyqVo4kWZtXjlWSRyqVo4kWZtXjlWSRyqVo4
kWZtXjlWSRyqVo4kWZtXjlWORy0iGRoleX6+oaXqfK4HVymdz9207OtY9pe6
qd8Ke6m9284CAe7V6Gt7+zCDu8Me229q97f7gI88fgW7ucq8W+x29tW85ONK
+6tNqOccdxC0eo6m9KX+T.6m5KX+79Wy9KV+wqya91e8K+1OdzU9Nd8mdk9i
64G+Msmyuf8bCj13KX+rA1OqunTdH1SwW5dB+zku7c8l97kOmt+h65fHu6uj
aiOH1SO9SB0SumRIGG+R1y+GNXScb7Kot4eVc+zSY70ey27O+xu8se+09i6i
2+rz+Su4iy07y93W9pW+ou7wG+xu8k+yu5yW+OcIe8299mP8cu+or9tu8SOU
6u+6+sF4q9cu482Be828puuuueO+u9S++Cmzp0+B
-----------end_max5_patcher-----------
</code></pre>

**Comment by bltzr** [comment]
I believe that loadtimes might be dramatically increased with nested models, mutliple devices, etc… 

could you cook us this kind of test patcher for testing purposes, @jln- ?

**Comment by avilleret** [comment]
sorry I should have been more clear...
as @bltzr said, only one patcher with only one leaf and only ossia object should be fast.

What makes things slower are :
  - the number of objects in a patcher (since currently we iterate over all of them)
  - the ramification ratio : the more there are patchers, bpatchers, poly~ containing ossia object the slowest it would be to register

**Comment by avilleret** [comment]
@jcelerier could you try if your patcher loads more quickly now ?
it should be necessary to implement such a mechanism after the rework of registration
let me know

**Comment by bltzr** [comment]
> it should be necessary to implement such a mechanism

you meant: "it should NOT be necessary", didn't you ?

**Comment by avilleret** [comment]
ping @jcelerier si tu sais retrouver ton patch qui mettait 5 minutes à charger, je ne l'ai plus :-)

**Comment by avilleret** [comment]
we need some very complex test patchers now to see if everthing is fine
hopefully objects will be registered only once now this should speed up things a lot
